pkcs5/src/pbes2/encryption.rs

```diff
@@ -178,9 +186,17 @@ impl EncryptionKey {
 
     /// Derive key using PBKDF2.
     fn derive_with_pbkdf2<D>(password: &[u8], params: &Pbkdf2Params<'_>, length: usize) -> Self
-    where
-        D: Update + BlockInput + FixedOutput + Reset + Default + Clone + Sync,
-        D::BlockSize: ArrayLength<u8>,
+        where
+            D: CoreProxy,
+            D::Core: HashMarker
+                + UpdateCore
+                + FixedOutputCore
+                + BufferKindUser<BufferKind = Eager>
+                + Default
+                + Clone
+                + Sync,
+            <D::Core as BlockSizeUser>::BlockSize: IsLess<U256>,
+            Le<<D::Core as BlockSizeUser>::BlockSize, U256>: NonZero,
```

Comment:
I know it's not pretty... But it's that we currently have. We could use [`SimpleHmac`](https://docs.rs/hmac/latest/hmac/struct.SimpleHmac.html), but it will be a bit less efficient.

Reply:
Ideally we would have trait aliases, but AFAIK they are far from being implemented and stabilized. An alternative approach could be introduction of a helper trait in the `hmac` crate with blanket impl for types satisfying the currently used bound, but it would require somewhat annoying passing of wrapped methods and associated types.